<launch>
  <!-- variables for pretrained cnn used for tracking -->
  <arg name="pretrained_weights" default="models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel" />
  <arg name="imagenet_mean" default="python/caffe/imagenet/imagenet_mean.binaryproto" />
  <arg name="deployment_prototxt" default="$(find kernelized_correlation_filters_gpu)/models/deploy.prototxt" />

  <!-- variables pretrained cnn used for regression network -->
  <arg name="regress_net_weights" default="nets/models/pretrained_model/tracker.caffemodel" />
  <arg name="regress_net_proto" default="nets/tracker.prototxt" />
  
  <!-- yaml file containing hyperparameters for kcf -->
  <arg name="kcf_params" default="$(find kernelized_correlation_filters_gpu)/models/kcf_params.yaml" />

  <!-- ros subscription topic names -->
  <arg name="image" default="/camera/image_rect" />
  <arg name="input_screen" default="$(arg image)/screenrectangle" />  
  <arg name="odom" default="/dji_sdk/odometry" />

  <arg name="init_image" default="/detector/output/image" />
  <arg name="init_rect" default="/detector/output/rectangle" />
  <arg name="init_odom" default="$(arg odom)" />

  <!-- parameters controlling respective behavious -->
  <arg name="rectangle_resizing_scale" default="1" />
  <arg name="downsize" default="1" />
  <arg name="headless" default="true" />
  <arg name="runtype_without_uav" default="true" />
  <arg name="init_from_detector" default="false" />
  
  <node pkg="kernelized_correlation_filters_gpu" type="kernelized_correlation_filters_gpu_node"
	name="kernelized_correlation_filters_gpu" output="screen" >
    <remap from="image" to="$(arg image)"/>
    <remap from="input_screen" to="$(arg input_screen)"/>
    <remap from="odom" to="$(arg odom)" />
    <param name="rectangle_resizing_scale" value="$(arg rectangle_resizing_scale)" />
    <param name="downsize" value="$(arg downsize)" />
    <param name="headless" value="$(arg headless)" />
    <param name="kcf_params" value="$(arg kcf_params)" />
    <param name="runtype_without_uav" value="$(arg runtype_without_uav)" />
    <param name="init_from_detector" value="$(arg init_from_detector)" />
   
    <remap from="init_image" to="$(arg init_image)" if="$(arg init_from_detector)" />
    <remap from="init_rect" to="$(arg init_rect)" if="$(arg init_from_detector)" />
    <remap from="init_odom" to="$(arg init_odom)" if="$(arg init_from_detector)" />

    <param name="pretrained_weights" value="$(arg pretrained_weights)" />
    <param name="model_prototxt" value="$(arg deployment_prototxt)" />
    <param name="imagenet_mean" value="$(arg imagenet_mean)" /> 

    <param name="regress_net_weights" value="$(arg regress_net_weights)" />
    <param name="regress_net_proto" value="$(arg regress_net_proto)" />
  </node>

  <group unless="$(arg headless)" >
    <node pkg="image_view2" type="image_view2"
    	  name="image_view2" output="screen">
      <remap from="image" to="$(arg image)"/>
    </node>
  </group>

</launch>
